config-provisioning/src/main/java/com/yahoo/config/provision/NodeType.java

```diff
@@ -88,6 +88,14 @@ public boolean canRun(NodeType type) {
         return childNodeTypes.contains(type);
     }
 
+    /** Returns the parent host type. */
+    public NodeType parentNodeType() {
```

Comment:
What's wrong with `NodeType::hostType`?

Reply:
It's pretty good.  It returns `this` if the type is already a host.  In our case I need the parent host, and it's wrong if the parent host type doesn't exist. :shrug: